couch2pg-entrypoint.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -e
+WAIT_THRESHOLD="${WAIT_THRESHOLD:-20}"
+SLEEP_SECONDS="${SLEEP_SECONDS:-5}"
+
+welcome_message(){
+  echo "Starting couch2pg process">&2
+}
+
+set_postgres_url(){
+  export POSTGRESQL_URL=postgres://$POSTGRES_USER_NAME:$POSTGRES_PASSWORD@$POSTGRES_DB:5432/$POSTGRES_DB_NAME
+  echo "Set postgres URL to $POSTGRESQL_URL" >&2
+}
+
+check_if_postgres_is_ready(){
+  #waiting for postgres
+  wait_count=0
+  until  pg_isready -q  -h $POSTGRES_DB -U $POSTGRES_USER_NAME --d $POSTGRES_DB_NAME
+  do
+    echo "Waiting for PostgreSQL..." >&2
+    wait_count=$((wait_count +1))
+    if [[ "$wait_count" -gt $WAIT_THRESHOLD ]]; then
+      echo "No  PostgreSQL DB Found" >&2
+      exit 1
+    fi
+    sleep $SLEEP_SECONDS
+  done
+echo "Postgres is ready moving on ...">&2
+}
+
+
+check_if_api_is_ready(){
```

Comment:
Related to one of Nick's comments, should this be: `check if couchdb is ready` or a mix of both? The flow is, couchdb is required here, but if you set up the endpoint as API, then API would check if couchdb is available.